pubnub/lib/src/dx/files/files.dart

```diff
@@ -100,28 +100,26 @@ class FileDx {
 
     var publishFileResult = PublishFileMessageResult();
 
-    if (s3Response.statusCode == 204) {
```

Comment:
Agree!! So If file upload call fails It will throw exception. due to https://github.com/pubnub/dart/blob/bbd67d227bc3e59a618e7fb90ddc312f55fff4da/pubnub/lib/src/networking/request_handler/html.dart#L139

And It won't proceed further for publish-file message.

And Exception message gives enough info about reason of failure.

Reply:
Oh yes, that is absolutely correct! Nice catch!